yaks/lib/yaks/format/json_api.rb

```diff
@@ -38,7 +38,7 @@ def serialize_resource(resource)
       def serialize_links(subresources)
         subresources.each_with_object({}) do |resource, hsh|
           next if resource.null_resource?
-          hsh[resource.type] = serialize_link(resource)
+          hsh[resource.rels.first] = serialize_link(resource)
```

Comment:
`rels` at the resource level means the relationship a resource has to its parent. Originally we stored subresources as a `{rel => resource}` hash, but some formats explicitly allow mutliple rels, so it was moved into the child resource, which I think makes for a better modeling in general.

Does that help you answer your question?

I'm mulling over some changes to make rels/identifiers in Yaks smarter in general, because different formats might represent things in different ways, e.g.

```
"http://api.example.com/rel/widgets"
"rel:widgets"
"widgets"
```

in that case a `rel` will be some object that knows how to represent itself as a full URI, a compacted CURIE form, just a "short name", etc. For a format like JSON-API where (I think) you would just use simple type/tag names instead of full URI's this would help. (It's also an important preparatory step to supporting RDF based stuff like JSON-LD).